src/emulator/auth/state.ts

```diff
@@ -631,6 +636,54 @@ export class AgentProjectState extends ProjectState {
     return true;
   }
 
+  get config() {
+    return this._config;
+  }
+
+  get blockingFunctionsConfig() {
+    return this._config.blockingFunctions;
+  }
+
+  set blockingFunctionsConfig(blockingFunctions: BlockingFunctionsConfig) {
+    this._config.blockingFunctions = blockingFunctions;
+  }
+
+  // TODO(lisajian): Once v2 API discovery is updated, type of update should be
+  // changed to Schemas["GoogleCloudIdentitytoolkitAdminV2Config"].
+  updateConfig(
+    update: Schemas["GoogleCloudIdentitytoolkitAdminV2Config"] & { usageMode?: UsageMode },
+    updateMask: string | undefined
+  ): Config {
+    if (update.usageMode) {
+      assert(update.usageMode !== UsageMode.USAGE_MODE_UNSPECIFIED, "Invalid usage mode provided");
```

Comment:
WDYT about moving request validation to `operation.ts` (following the convention of other handlers)? Ideally, `state.ts` should only need to handle valid arguments.